rust_src/src/crypto/mod.rs

```diff
@@ -356,13 +356,13 @@ fn secure_hash(
         LispObject::constant_nil(),
         LispObject::constant_nil(),
     );
-    _secure_hash(hash_alg(algorithm), input.as_slice(), binary.is_nil())
+    _secure_hash(&hash_alg(algorithm), input.as_slice(), binary.is_nil())
 }
 
-fn _secure_hash(algorithm: HashAlg, input: &[u8], hex: bool) -> LispObject {
+fn _secure_hash(algorithm: &HashAlg, input: &[u8], hex: bool) -> LispObject {
```

Comment:
This is not an improvement IMO. I guess the lint triggeres because `HashAlg` is not `Copy` - try adding `#[derive(Copy)]` there.

Reply:
As a side note: there is actually a small memory improvement if a copy/move is used in this case:

```
assert_eq!(1, std::mem::size_of::<HashAlg>());
assert_eq!(8, std::mem::size_of::<&HashAlg>());
```

However, I think in this context it will not have a huge impact.